### ISSUE TITLE
Converter producing invalid worlds

### ISSUE THREAD

**Issue by Anonymous941** [Cracko298/mc3dslib#5]
When trying to use this on the Skyblock world, the result is an invalid world

**Comment by Cracko298** [comment]
Yes, this is an Experimental Feature and is not Reccomended for usage until further research is done.

**Comment by Anonymous941** [comment]
I'm using `leveldb-mcpe` to parse this, and here are the problems, in order:
- The `CURRENT` file needs a newline.  This can easily be fixed, see #6
- The CRC is not correct.  Can be fixed by patching out the check, but generating a proper CRC shouldn't be too hard.  I'll see if I can add it later
- The log file is not in the correct format.  For some reason, the script copies `db/vdb/newindex.vdb` to `db/MANIFEST-000001`, despite these files having no resemblance I could figure out.

This is where I'm currently stuck, as it won't continue without a valid log file, and patching out the check just parses a blank world with no chunks.  I don't understand what it even is, but it seems to be used partially for crash recovery - [here is the relevant documentation](https://github.com/google/leveldb/blob/main/doc/log_format.md)

**Comment by Cracko298** [comment]
Mc3ds isn't a traditional bedrock save it is close enough todo some sort of conversion and renaming

**Comment by Anonymous941** [comment]
Where is the log file?  `newindex.vdb` doesn't seem even close, and I can't find any other files that seem to have the correct structure

**Comment by Cracko298** [comment]
The 3ds Edition doesn't contain any log files. They representative of each other around PE 1.0.15 (not bedrock).

**Comment by Anonymous941** [comment]
The library requires one, is there a way around this?

**Comment by Cracko298** [comment]
You can modify it out of the module, but no guarantee it'll work

**Comment by Anonymous941** [comment]
How did you get the void world in Skyblock without converting a save?  I'm trying to make a void nether, but I can't use any save editors because I can't convert it into a Bedrock save

**Comment by Cracko298** [comment]
I manually translated the Bytes by hand and used a pre-existing world as a template.

It's tedious and takes roughly 100 hours to completely convert a world this way.

**Comment by Anonymous941** [comment]
Wow, that's insane dedication, no wonder you didn't fix Cracko298/MC3DS-Skyblock-Plus#2 (that could take years lol).  I can try to make a program to automate this, how did you figure out the blocks?  Which files are they stored in?

**Comment by Cracko298** [comment]
If you need to know why that happens (for Skyblock+). It happens because the world doesn't register the VOID as valid when saving and assumes it is wrong or bad data.

As for where the actual data is stored, you can look into CDB files. Index files do NOT contain any world data and just point to where the slot files are located inside of the folder and naming scheme.

**Comment by Cracko298** [comment]
If you would like to join us, we have a bounty for finding a full proof method and providing code to convert to/from mc3ds and Java or Bedrock in our community server.

The Bounty is $1000 for software that can convert it or $200 for a writeup of said conversion.

**Comment by Anonymous941** [comment]
> If you would like to join us, we have a bounty for finding a full proof method and providing code to convert to/from mc3ds and Java or Bedrock in our community server.
> 
> The Bounty is $1000 for software that can convert it or $200 for a writeup of said conversion.

$1000??  ~~If it's gone unclaimed all this time~~ If it's this hard, I'm probably not skilled enough write it completely (or someone probably would have already), but I'd be happy to help to the best of my ability.  I've been in the server for a week or so actually, my username is `phoneticteacher`

**Comment by Cracko298** [comment]
It just was posted today, but the bounty ends in one month (09/29/2024l.
